### ISSUE TITLE
Unclear line of code to check for constant parameter block in BlockJacobianWriter::CreateJacobian()

### ISSUE THREAD

**Issue by jwmak** [ceres-solver/ceres-solver#971]
Inside the function BlockJacobianWriter::CreateJacobian() at line 186 in block_jacobian_writer.cc, the following line checks if the parameter block is constant:

if (residual_block->parameter_blocks()[j]->index() != -1)

For improved clarity, this line should be changed to:

if (!residual_block->parameter_blocks()[j]->IsConstant())

**Comment by sandwichmaker** [comment]
Yes this is some ancient code, and it is likely that at this point parameter_block did not have an IsConstant method, I will look into modernizing it.
